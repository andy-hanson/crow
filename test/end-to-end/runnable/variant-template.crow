main void()
	info log "{1.x f}"
	info log "{2::nat.x2 g}"
	()

v[t] variant mut
x[t] variant-member(t v, t)
f nat(a nat v)
	match a
	as x value
		value
	else
		0

v2 variant mut
x2[t] variant-member(v2, t)
g nat(a v2)
	match a
	as x2 value nat
		value
	else
		0
