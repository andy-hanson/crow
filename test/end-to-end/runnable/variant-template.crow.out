info: 1
info: 2
